net/Socket.hpp

```diff
@@ -746,9 +747,8 @@ class SocketPoll
     void addCallback(const CallbackFn& fn)
     {
         std::lock_guard<std::mutex> lock(_mutex);
-        bool wasEmpty = _newCallbacks.empty();
         _newCallbacks.emplace_back(fn);
-        if (wasEmpty)
+        if (!_newSockets.empty() || !_newCallbacks.empty())
```

Comment:
This condition is not helpful because it must always be true (we just inserted an item into `_newCallbacks`). However, the old solution was smart, so we should bring it back (and apply it in the `insertNewSocket()` case too).

You see, normally when we insert into either `_newCallbacks` or `_newSockets`, we want to `wakeup()`. But this `wakeup()` is expensive, because it needs to write into the pipe. The good news is that we can optimize this when we know `wakeup()` is already called. The previous code here did that: it checked if `_newCallbacks.empty()` *before* inserting and called `wakeup()` only when it was empty. Meaning, it only calls `wakeup()` the first time we insert something, but after that, any further insertions don't need separate calls to `wakeup()` since that needs to be done only once (util we read and empty `_newCallbacks` and `_newSockets`.

So, the solution here and in the `insertNewSocket()` function is this:
```
const bool wasEmpty = _newSockets.empty() && _newCallbacks.empty();
_newCallbacks.emplace_back(fn);
if (wasEmpty)
    wakeup();
```
(Same solution for the `insertNewSocket()` case, but change the second line accordingly.)